### DISCUSSION TITLE
Not Found Error

### DISCUSSION THREAD

**Discussion by mounir788** [vercel/community#6098] [Help] [answered]
### Summary

I deployed my react app and added a domain to it, well when visit the app and reload the page it shows this error:
404: NOT_FOUND
Code: NOT_FOUND
ID: cdg1::lbxxb-1710206189685-64046aede2c7

I have to reenter the url and hit search to back to the app and remove this error

### Example

https://www.mounirr.com/

### Steps to Reproduce

404: NOT_FOUND
Code: NOT_FOUND
ID: cdg1::lbxxb-1710206189685-64046aede2c7

I tried the steps in the vercel documentation for this error but still the same

**Comment by amyegan** [comment]
Hey @mounir788. It looks like this problem is coming from some routing configuration in the app. 

When I open https://www.mounirr.com I am redirected to `/home`. If I try to refresh from there, or directly visit https://www.mounirr.com/home, I get a 404 error. The same thing happens for `/about` and `/contact`.

**Reply by mounir788** [reply]
I fixed it now, I didn't expect that this would be the problem.
This is the old code that causes the problem:

```
 <Routes location={location} key={location.pathname}>
        <Route element={<AppLatyout />}>
          <Route index element={<Navigate replace to="home" />} />
          <Route path="home" element={<Home />} />
          <Route path="about" element={<About />} />
          <Route path="work" element={<Work />} />
          <Route path="contact" element={<Contact />} />
          <Route path="work/:workName" element={<WorkDetails />} />
        </Route>
</Routes>
```
      
      And this is the new one that fixed the problem:

```
      <Routes location={location} key={location.pathname}>
        <Route element={<AppLatyout />}>
          <Route index element={<Home />} />
          <Route path="home" element={<Home />} />
          <Route path="about" element={<About />} />
          <Route path="work" element={<Work />} />
          <Route path="contact" element={<Contact />} />
          <Route path="work/:workName" element={<WorkDetails />} />
        </Route>
      </Routes>
```
      
      Thanks for your help.

**Reply by mounir788** [reply]
But this fixed it to the home page only it still the same problem with the other pages

**Reply by mounir788** [reply]
Here is the repo link if you can take a look at the code and help me please:
https://github.com/mounir788/Portfolio

Thanks in advance

**Reply by sao-coding** [reply, marked as answer]
![image](https://github.com/vercel/community/assets/58339015/e1ad105f-f801-4cfc-8b6f-4d3f0bd55d8e)
All pages are in this js file
So you need to create vercel.json and point all paths to index.html
```json
{
    "rewrites": [
        {
            "source": "/(.*)",
            "destination": "/index.html"
        }
    ]
}
```

**Reply by mounir788** [reply]
@sao-coding  Thanks a lot! it works fine now
